packages/typespec-azure-resource-manager/src/private.decorators.ts

```diff
@@ -359,3 +359,18 @@ export function isAzureResource(program: Program, resourceType: Model): boolean
   const isResourceBase = program.stateMap(ArmStateKeys.azureResourceBase).get(resourceType);
   return isResourceBase ?? false;
 }
+
+/**
+ * Please DO NOT USE in RestAPI specs.
+ * Internal decorator that deprecated direct usage of `x-ms-client-flatten` OpenAPI extension.
+ * It will programatically enabled/disable client flattening with @flattenProperty with autorest
```

Comment:
@allenjzhang i do not see where to set `@flattenProperty` decorator, but only see the set of openapi2 `x-ms-client-flatten`. am i miss sth?